### PR TITLE
Update for pino@9

### PR DESCRIPTION
**Pull request by jsumners**
Part of https://github.com/pinojs/pino/issues/1762.

### CODE REVIEW

File: test/base.test.js [hunk 1]
```diff
@@ -382,8 +384,9 @@ test('close with promises', ({ same, plan, pass }) => {
   stream.end()
 })
 
-test('support Transform streams', ({ same, plan, error }) => {
-  plan(7)
+test('support Transform streams', (t) => {
+  // TODO: why isn't `plan: 7` counted correctly?
+  const { deepEqual, ifError } = tspl(t, { plan: 7 })
```
Comment (jsumners) [inline test/base.test.js:389]: I have no idea why this is not getting counted correctly. I can step through all of the assertions and see `actual === plan` is true in `tspl`. @mcollina do you have any clues?

**Comment by mcollina** [comment]
CI doesn't seem happy

**Comment by jsumners** [comment]
> CI doesn't seem happy

Yeah. See my inline review comment.

**Comment by jsumners** [comment]
I give up for now. `node:test` in v16 doesn't include `t.after` so `tspl` isn't usable. But we need `tspl` to assert the planned number of assertions. This PR has completely convinced me that the functionality belongs in `node:test` and the reasoning of "we want all assertion libraries to work" is a cop out.

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**